mmdet/integration/nncf/compression.py

```diff
@@ -0,0 +1,210 @@
+import pathlib
+import torch
+from functools import partial
+
+from mmdet.utils import get_root_logger
+from .utils import (check_nncf_is_enabled, get_nncf_version, is_nncf_enabled,
+                    load_checkpoint)
+
+if is_nncf_enabled():
+    try:
+        from nncf import (NNCFConfig, create_compressed_model,
+                          register_default_init_args)
+        from nncf.dynamic_graph.patch_pytorch import nncf_model_input
+        from nncf.initialization import InitializingDataLoader
+        from nncf.nncf_network import NNCFNetwork
+
+        class_InitializingDataLoader = InitializingDataLoader
+    except ImportError:
+        raise RuntimeError(
+            'Cannot import the standard functions of NNCF library '
+            '-- most probably, incompatible version of NNCF. '
+            'Please, use NNCF version pointed in the documentation.')
+else:
+    class DummyInitializingDataLoader:
+        pass
+
+
+    class_InitializingDataLoader = DummyInitializingDataLoader
+
+
+class MMInitializeDataLoader(class_InitializingDataLoader):
+    def get_inputs(self, dataloader_output):
+        # redefined InitializingDataLoader because
+        # of DataContainer format in mmdet
+        kwargs = {k: v.data[0] for k, v in dataloader_output.items()}
+        return (), kwargs
+
+    # TODO: not tested; need to test
```

Comment:
have you tested?

Reply:
yes. Actually this method is never used. It can be removed

Reply:
Fixed. Thanks Aleksei for change